### DISCUSSION TITLE
Persist XP Tracker state across client restarts when logged in

### DISCUSSION THREAD

**Discussion by awdavis** [runelite/runelite#15561] [Ideas]
Similar to how the Loot Tracker preserves data across restarts of the RuneLite application when you are logged in (e.g. #4671), it would be very helpful if the XP Tracker did the same until manually cleared (or if there was a setting to control the behavior of whether data persists across application restarts).

Right now I am frequently faced with a "RuneScape has been updated! Please reload this page." message, which seems to require restarting the RuneLite application (per issue #6405 Refresh OSRS Within Client). Doing this resets my XP Tracker data, even while logged in to RuneLite, and I don't see any option to preserve it.

![screenshot](https://user-images.githubusercontent.com/2943788/187008033-83f4de56-d442-4356-8459-4f25b755cbb3.png)

**Comment by Marstead** [comment]
I also have a use case that would benefit from saving XP Tracker information across settings. I brought this up in Discord and Adam mentioned that there might be a problem with pausing if it has been disabled, even across logouts. I'd propose that if the RL session ends, the tracker should auto pause based on the last session end timestamp and resume based on the next session start timestamp.
